algorithms/HPdata/R/db2dframes.R

```diff
@@ -0,0 +1,384 @@
+            vertica_nodes <- sqlQuery(db_connect, qryString)
+            if(!is.data.frame(vertica_nodes))
+              stop(paste("Error in Vertica:", paste(vertica_nodes, sep="", collapse="")))
+
+            #get parameter string
+            udx_param <- .getUDxParameterLocalStr(vertica_nodes)
+            if(!udx_param$success){
+              if(udx_param$error_code=="ERR02") {
+                stop(udx_param$parameter_str)
+              } else {
+                udx_param <- .getUDxParameterUniformStr() 
+                type <- "uniform"
+                if(!udx_param$success) {
+                  stop(udx_param$parameter_str)
+                } 
+              }
+            } else {
+              type <- "local"
+            }
+          } else if(as.character(loadPolicy) == "uniform") {
+            udx_param <- .getUDxParameterUniformStr()
+            type <- "uniform"
+            if(!udx_param$success) {
+              stop(udx_param$parameter_str)
+            } 
+          } else {
+            stop(paste("Invalid data load policy selection", as.character(loadPolicy)))
+          }
+
+          #disable retry using MaxQueryRetries.
+          #sqlQuery(db_connect, "select set_vertica_options('BASIC', 'DISABLE_ERROR_RETRY');")
+          retries_allowed <- sqlQuery(db_connect, "select get_config_parameter('MaxQueryRetries');")
+          sqlQuery(db_connect, "select set_config_parameter('MaxQueryRetries', 0);")
+
+          #issue UDx query
+          qryString <- paste("select ExportToDistributedR(", columns, " USING PARAMETERS DR_worker_info='", udx_param$parameter_str,"', DR_partition_size=", partition_size, ", data_distribution_policy='", type, "') over(PARTITION BEST) from ", tableName, sep="")
+         
+          message(paste("Loading total", nRows, "rows from", tableName, "from Vertica with approximate partition of", partition_size,"rows")) 
+          load_result <- sqlQuery(db_connect, qryString)
+          if(!is.data.frame(load_result)) {
+            sqlQuery(db_connect, paste("select set_config_parameter('MaxQueryRetries', ", retries_allowed, ");"))
+            error_msg <- .Call("HandleUDxError", load_result, PACKAGE="distributedR")
+            stop(error_msg)
+          }
+
+          #clear retry
+          #sqlQuery(db_connect, "select clr_vertica_options('BASIC', 'DISABLE_ERROR_RETRY');")
+          sqlQuery(db_connect, paste("select set_config_parameter('MaxQueryRetries', ", retries_allowed, ");"))
+          
+          #get loader status from distributedR workers
+          result <- .getLoaderResult(load_result)
+          if(!is.list(result))
+            stop(result)
+
+          dResult <- .vertica.dframes(result, nResponses, nPredictors, pred_columns, resp)
+
+          }, interrupt = function(e) {}
+           , error = function(e) {
+             .vertica.connector(e)
+          } , finally = { 
+             stopDataLoader()
+          })
+
+    # end of verticaConnector
```

Comment:
That is almost correct. I have also changed the column type validation to be similar to db2dframe.